python/src/opendp/extras/mbi/_table/__init__.py

```diff
@@ -0,0 +1,548 @@
+            raise ValueError("model must be a MarkovRandomField")
+
+        for col in self.keys:
+            self.keys[col] = pl.Series(col, self.keys[col])
+        for col in self.cuts:
+            self.cuts[col] = pl.Series(col, self.cuts[col])
+
+        key_shape = set((c, len(k)) for c, k in self.keys.items())
+        if set(self.model.domain.config.items()) != key_shape:
+            raise ValueError(
+                "Model domain must match key attrs and sizes.\n"
+                f"  Model shape: {self.model.domain}\n"
+                f"  Key shape:   {Domain.fromdict(dict(key_shape))}"
+            )
+
+        for col, cutset in self.cuts.items():
+            if col not in self.keys:
+                raise ValueError(f'"{col}" in cuts is not present in keys')
+            if len(self.keys[col]) - 1 != len(cutset):
+                msg = f'"{col}" keyset length ({len(self.keys[col])}) must be one greater than the number of cuts ({len(cutset)})'
+                raise ValueError(msg)
+
+    def synthesize(
+        self, rows: Optional[int] = None, method: Literal["round", "sample"] = "round"
+    ):
+        """Generate synthetic data that is consistent with the contingency table.
+
+        :param rows: if not set, approximately matches number of rows in original data
+        :param method: "round" rounds to projection, "sample" samples from densities
+        """
+        import polars as pl  # type: ignore[import-not-found]
+        from mbi import MarkovRandomField  # type: ignore[import-untyped,import-not-found]
+
+        model = cast(MarkovRandomField, self.model)
+
+        indices = model.synthetic_data(rows, method).df
+        data = {
+            c: _deindex(indices[c].to_numpy(), self.keys[c], cuts=self.cuts.get(c))
+            for c in self.keys
+        }
+        return pl.DataFrame(data)
+
+    def project(self, attrs: str | Sequence[str]):
+        """Return counts corresponding to each combination of self.keys in `attrs` in an ndarray.
+
+        :param attrs: attributes to preserve. All other attributes are marginalized.
+        """
+        from mbi import MarkovRandomField  # type: ignore[import-untyped,import-not-found]
+
+        model = cast(MarkovRandomField, self.model)
+        return model.project(attrs).values
+
+    def project_melted(self, attrs: str | Sequence[str]):
+        """Return counts corresponding to each combination of self.keys in `attrs` in a dataframe.
+
+        :param attrs: attributes to preserve. All other attributes are marginalized.
+        """
+        import polars as pl
+        import numpy as np
+
+        attrs = [attrs] if isinstance(attrs, str) else list(attrs)
+
+        lengths_arr = np.asarray(self.project(attrs).ravel())
+        lengths = pl.Series("len", lengths_arr).to_frame()
+
+        return row_major_order(self.keys[attr] for attr in attrs).hstack(lengths)
+
+    def std(self, attrs: str | Sequence[str]) -> float:
```

Comment:
```suggestion
    def std(self, attrs: Sequence[str]) -> float:
```

Slight preference to keep the type simple: Not a burden to ask the user to wrap a single element in a list.